### ISSUE TITLE
QBtn.click throws error when no event passed to function

### ISSUE THREAD

**Issue by holtalanm** [quasarframework/quasar#5469]
**Describe the bug**
According to the type definitions for QBtn, the `click` function takes an _optional_ parameter of the event that triggered the click.  Even though the definition says it is optional, `QBtn.click()` will throw an error if there is nothing passed in as a parameter.

Example:
`button.click()` throws an error, while `button.click({})` does not. 

**Codepen/jsFiddle/Codesandbox (required)**
https://codepen.io/holtalanm/pen/LYYQBzW?&editable=true&editors=101

**To Reproduce**
1. Click on `Click Causing Error`
2. See alert indicating an error was thrown
3. Click on `Click With No Error`
4. See notification saying the hidden button was programmatically clicked.

**Expected behavior**
The `click` function should not throw an error when there are no arguments passed to it.

**Screenshots**
from the type definition:
```ts
export interface QBtn extends Vue {
    /**
     * Size in CSS units, including unit name or standard size name (xs|sm|md|lg|xl)
     */
    size? : string
    /**
     * Configure material ripple (disable it by setting it to 'false' or supply a config object)
     */
    ripple? : boolean | any
    /**
     * Define the button HTML DOM type
     */
    type? : string
    /**
     * Equivalent to Vue Router <router-link> 'to' property
     */
    to? : string | any
    /**
     * Equivalent to Vue Router <router-link> 'replace' property
     */
    replace? : boolean
    /**
     * The text that will be shown on the button
     */
    label? : string | number
    /**
     * Icon name following Quasar convention; make sure you have the icon library installed unless you are using 'img:' prefix
     */
    icon? : string
    /**
     * Icon name following Quasar convention; make sure you have the icon library installed unless you are using 'img:' prefix
     */
    iconRight? : string
    /**
     * Use 'outline' design
     */
    outline? : boolean
    /**
     * Use 'flat' design
     */
    flat? : boolean
    /**
     * Remove shadow
     */
    unelevated? : boolean
    /**
     * Applies a more prominent border-radius for a squared shape button
     */
    rounded? : boolean
    /**
     * Use 'push' design
     */
    push? : boolean
    /**
     * Applies a glossy effect
     */
    glossy? : boolean
    /**
     * Makes button size and shape to fit a Floating Action Button
     */
    fab? : boolean
    /**
     * Makes button size and shape to fit a small Floating Action Button
     */
    fabMini? : boolean
    /**
     * Color name for component from the Quasar Color Palette
     */
    color? : string
    /**
     * Overrides text color (if needed); color name from the Quasar Color Palette
     */
    textColor? : string
    /**
     * Avoid turning label text into caps (which happens by default)
     */
    noCaps? : boolean
    /**
     * Avoid label text wrapping
     */
    noWrap? : boolean
    /**
     * Dense mode; occupies less space
     */
    dense? : boolean
    /**
     * Tabindex HTML attribute value
     */
    tabindex? : number | string
    /**
     * Label or content alignment
     */
    align? : string
    /**
     * Stack icon and label vertically instead of on same line (like it is by default)
     */
    stack? : boolean
    /**
     * When used on flexbox parent, button will stretch to parent's height
     */
    stretch? : boolean
    /**
     * Put button into loading state (displays a QSpinner -- can be overriden by using a 'loading' slot)
     */
    loading? : boolean
    /**
     * Put component in disabled mode
     */
    disable? : boolean
    /**
     * Makes a circle shaped button
     */
    round? : boolean
    /**
     * Percentage (0.0 < x < 100.0); To be used along 'loading' prop; Display a progress bar on the background
     */
    percentage? : number
    /**
     * Progress bar on the background should have dark color; To be used along with 'percentage' and 'loading' props
     */
    darkPercentage? : boolean
    /**
     * Emulate click on QBtn
     * @param evt JS event object
     */
    click (evt? : any): void
}
```

**Platform (please complete the following information):**
OS: Windows 10
Browsers: Firefox 70

**Comment by metalsadman** [comment]
use `this.$refs.theButton.$el.click()` https://codepen.io/metalsadman/pen/vYYdPJe?&editable=true&editors=101

**Comment by rstoenescu** [comment]
Fix will go into "quasar" v1.4.0.
